controller/program/src/instructions/start_rebalance.rs

```diff
@@ -0,0 +1,491 @@
+    let accounts_start = current;
+    let accounts_size = num_accounts as usize * 33;
+    if current + accounts_size > data.len() {
+        return Err(INVALID_ACCOUNT_DATA.into());
+    }
+
+    // Extract account pubkeys
+    let mut account_pubkeys = Vec::with_capacity(num_accounts as usize);
+    for i in 0..num_accounts as usize {
+        let acc_start = accounts_start + i * 33 + 1; // +1 to skip meta byte
+        if acc_start + 32 > data.len() {
+            return Err(INVALID_ACCOUNT_DATA.into());
+        }
+        let mut pubkey = [0u8; 32];
+        pubkey.copy_from_slice(&data[acc_start..acc_start + 32]);
+        account_pubkeys.push(pubkey);
+    }
+
+    current += accounts_size;
+
+    // Read program_id (32 bytes)
+    if current + 32 > data.len() {
+        return Err(INVALID_ACCOUNT_DATA.into());
+    }
+    let program_id_start = current;
+    current += 32;
+
+    // Read data_len
+    if current + 2 > data.len() {
+        return Err(INVALID_ACCOUNT_DATA.into());
+    }
+    let data_len = u16::from_le_bytes([data[current], data[current + 1]]);
+    current += 2;
+
+    // Read first byte of instruction data (discriminator)
+    let discm = if data_len > 0 && current < data.len() {
+        data[current]
+    } else {
+        0
+    };
+
+    let mut program_id = [0u8; 32];
+    program_id.copy_from_slice(&data[program_id_start..program_id_start + 32]);
+
+    Ok((program_id, discm, account_pubkeys))
+}
+
+const END_REBALANCE_IX_DST_LST_MINT_INDEX: usize = 4;
+
+/// Verify that an EndRebalance instruction exists after the current instruction with the expected destination mint
+#[inline]
+fn verify_end_rebalance_exists(
+    instructions_data: &[u8],
+    expected_inp_lst_mint: &[u8; 32],
+) -> Result<(), ProgramError> {
+    let current_idx = load_current_index(instructions_data);
+
+    let mut next = current_idx
+        .checked_add(1)
+        .ok_or(Inf1CtlCustomProgErr(Inf1CtlErr::MathError))?;
+    let mut found_end_rebalance = false;
+
+    // TODO: better way to find the EndRebalance instruction?
+    while let Ok((program_id, discm, accounts)) =
+        load_instruction_at(next as usize, instructions_data)
+    {
+        if program_id == ID && discm == END_REBALANCE_IX_DISCM {
+            // Verify the EndRebalance has enough accounts
+            if accounts.len() <= END_REBALANCE_IX_DST_LST_MINT_INDEX {
+                return Err(Inf1CtlCustomProgErr(Inf1CtlErr::NoSucceedingEndRebalance).into());
+            }
+
+            // Verify destination mint matches
+            if accounts[END_REBALANCE_IX_DST_LST_MINT_INDEX] != *expected_inp_lst_mint {
+                return Err(Inf1CtlCustomProgErr(Inf1CtlErr::NoSucceedingEndRebalance).into());
+            }
+
+            found_end_rebalance = true;
+            break;
+        }
+        next = next
+            .checked_add(1)
+            .ok_or(Inf1CtlCustomProgErr(Inf1CtlErr::MathError))?;
+    }
```

Comment:
this is pretty close to the original impl: https://github.com/igneous-labs/S/blob/9fc17034f7b5376f27a6925baa1e65af05843cba/programs/s-controller/src/processor/start_rebalance.rs#L196-L232

But to maybe "improve" on it, what you want to do here is use all the library functions we have defined
- use `EndRebalanceIxAccs` to help you find `dst_lst_mint` instead of using raw indices